src/cmd/run.rs

```diff
@@ -275,37 +287,36 @@ async fn poll_integration<D: Database>(
-        // if an error is encountered reading the stream, we don't send partial lists.
-        if references.is_empty() {
-            info!("No changes to '{integration}'");
-        }
-        for reference in references {
-            let job = ScanGitVCSReference::new(integration, &reference);
-            sender.send(&job).await.change_context(Error::TaskEnqueue)?;
-
-            info!("Enqueued task to scan '{integration}' at '{reference}'");
-        }
+    // We sink the references here instead of during the stream so that
+    // if an error is encountered reading the stream, we don't send partial lists.
+    if references.is_empty() {
+        info!("No changes to '{integration}'");
+    }
+    for reference in references {
+        let job = ScanGitVCSReference::new(integration, &reference);
+        sender.send(&job).await.change_context(Error::TaskEnqueue)?;
 
-        // Now wait for the next poll time.
-        // The fact that we poll, _then_ wait for the poll time, means that the actual
-        // time at which the poll occurs will slowly creep forward by whatever time
-        // it takes to perform the poll.
-        //
-        // This is considered okay, because we're interpreting "poll interval" as
-        // "poll at most this often"; i.e. it's used primarily as a rate limiting feature
-        // than as a "track changes this often" feature.
-        //
-        // The alternative would be starting the clock at the top of the loop,
-        // which while doable could (in the worst case) allow for endlessly
-        // polling if the poll takes at least as long as the interval.
-        //
-        // If we decide to make polling more consistent, [`tokio::time::interval`]
-        // is most likely the correct way to implement it.
-
-        info!("Next poll interval for '{integration}' in {poll_interval:?}");
-        tokio::time::sleep(poll_interval).await;
+        info!("Enqueued task to scan '{integration}' at '{reference}'");
     }
+    // Now wait for the next poll time.
+    // The fact that we poll, _then_ wait for the poll time, means that the actual
+    // time at which the poll occurs will slowly creep forward by whatever time
+    // it takes to perform the poll.
+    //
+    // This is considered okay, because we're interpreting "poll interval" as
+    // "poll at most this often"; i.e. it's used primarily as a rate limiting feature
+    // than as a "track changes this often" feature.
+    //
+    // The alternative would be starting the clock at the top of the loop,
+    // which while doable could (in the worst case) allow for endlessly
+    // polling if the poll takes at least as long as the interval.
+    //
+    // If we decide to make polling more consistent, [`tokio::time::interval`]
+    // is most likely the correct way to implement it.
+
+    info!("Next poll interval for '{integration}' in {poll_interval:?}");
+    tokio::time::sleep(poll_interval).await;
```

Comment:
Please move this sleep out into the loop to make it more obvious (and make this function more general).

Please apply this feedback to any similar loop with a function call and a sleep.